android/library/src/main/java/com/polidea/multiplatformbleadapter/BleModule.java

```diff
@@ -0,0 +1,1322 @@
+
+    public BleModule(Context context) {
+        this.context = context;
+        bluetoothManager = (BluetoothManager) context.getSystemService(Context.BLUETOOTH_SERVICE);
+        bluetoothAdapter = bluetoothManager.getAdapter();
+    }
+
+    // Lifecycle -----------------------------------------------------------------------------------
+
+    @Override
+    public void createClient(String restoreStateIdentifier,
+                             OnEventCallback<String> onAdapterStateChangeCallback,
+                             OnEventCallback<Integer> onStateRestored) {
+        rxBleClient = RxBleClient.create(context);
+        adapterStateChangesSubscription = monitorAdapterStateChanges(context, onAdapterStateChangeCallback);
+
+        // We need to send signal that BLE Module starts without restored state
+        if (restoreStateIdentifier != null) {
+            onStateRestored.onEvent(null);
+        }
+    }
+
+    @Override
+    public void destroyClient() {
+        // Subscriptions
+        if (adapterStateChangesSubscription != null) {
+            adapterStateChangesSubscription.unsubscribe();
+            adapterStateChangesSubscription = null;
+        }
+        if (scanSubscription != null && !scanSubscription.isUnsubscribed()) {
+            scanSubscription.unsubscribe();
+            scanSubscription = null;
+        }
+        transactions.removeAllSubscriptions();
+        connectingDevices.removeAllSubscriptions();
+
+        // Caches
+        discoveredServices.clear();
+        discoveredCharacteristics.clear();
+        connectedDevices.clear();
+        discoveredDevices.clear();
+
+        // Clear client
+        rxBleClient = null;
+        IdGenerator.clear();
+    }
+
+
+    @Override
+    public void enable(final String transactionId,
+                       final OnSuccessCallback<Void> onSuccessCallback,
+                       final OnErrorCallback onErrorCallback) {
+        changeAdapterState(
+                RxBleAdapterStateObservable.BleAdapterState.STATE_ON,
+                transactionId,
+                onSuccessCallback,
+                onErrorCallback);
+    }
+
+    @Override
+    public void disable(final String transactionId,
+                        final OnSuccessCallback<Void> onSuccessCallback,
+                        final OnErrorCallback onErrorCallback) {
+        changeAdapterState(
+                RxBleAdapterStateObservable.BleAdapterState.STATE_OFF,
+                transactionId,
+                onSuccessCallback,
+                onErrorCallback);
+    }
+
+    @BluetoothState
+    @Override
+    public String getCurrentState() {
+        if (!supportsBluetoothLowEnergy()) return BluetoothState.UNSUPPORTED;
+        if (bluetoothManager == null) return BluetoothState.POWERED_OFF;
+        return mapNativeAdapterStateToLocalBluetoothState(bluetoothAdapter.getState());
+    }
+
+    @Override
+    public void startDeviceScan(String[] filteredUUIDs,
```

Comment:
For me it's a cosmetic issue at this moment (since we didn't want to perform a big refactor in this codebase and also iOS codebase looks the same)and I will leave the decision for others.

Reply:
I agree with @Cierpliwy, but this shouldn't be addressed in this PR. Create an issue for it, @bartoszwilk.